docs/spire-multi-cluster.md

```diff
@@ -0,0 +1,311 @@
+# Deploying Tornjak in multi-cluster environment
+This document describes the steps required to extend the existing
+Tornjak/SPIRE Server to support multiple clusters.
+
+This assumes the Tornjak Server is already deployed as described in
+[SPIRE with Helm](./spire-helm.md) or [SPIRE on OpenShift](./spire-on-openshift.md)
+documents. If this is a new install, please follow the steps there for setting
+a multi-cluster support.
+
+Tornjak Server and all the remote cluster must use the same trust domain. E.g:
+```
+trust_domain = "openshift.space-x.com"
+```
+
+For this example we would use the following clusters:
+* `space-x01` - local cluster, with public Ingress access. Here we run the Tornjak/SPIRE
+Server under `tornjak` namespace, and local SPIRE agents.
+* `space-x02` - remote cluster (OpenShift)
+* `kube-01` - remote cluster (standard Kubernetes)
+
+## Step 1. Gather all KUBECONFIGs from the remote servers.
+SPIRE server attests the remote agents by verifying the KUBECONFIG configuration.
+Therefore, for every remote cluster,
+other then the cluster hosting the Tornjak,
+we need KUBECONFIG information.
+
+Gather the portable KUBECONFIGs for every remote cluster
+and output them into individual files
+using  `kubectl config view --flatten`.
+Make sure to use `--flatten` flag as it makes the output portable:
+
+```
+export KUBECONFIG=....
+export CLUSTERNAME=....
+mkdir /tmp/kubeconfigs
+kubectl config view --flatten > /tmp/kubeconfigs/$CLUSTERNAME
+```
+
+In our example we will have the following:
+```
+/tmp/kubeconfigs/space-x02
+/tmp/kubeconfigs/kube-01
+```
+
+Then using these individual files,
+create one secret in Tornjak hosting cluster, in `tornjak` namespace.
+_Note_: don't use special characters, stick to alphanumerics.
+
+```console
+kubectl -n tornjak create secret generic kubeconfigs --from-file=/tmp/kubeconfigs
+```
+
+## Step 2. Update SPIRE Server configuration.
+Update the SPIRE server configuration, by extending the NodeAttestor "k8s_psat"
+arguments.
+
+For all remote servers we need `kube_config_file` that contains its cluster KUBECONFIG.
+We don't need this for the local cluster.
+
+We assume all the SPIRE agent are deployed in `spire` namespace
+under `spire-agent` service-account,
+otherwise change the configuration accordingly:
+
+Update the spire-server ConfigMap:
+
+```console
+kubectl -n tornjak edit configmap spire-server
+```
+
+```yaml
+NodeAttestor "k8s_psat" {
+  plugin_data {
+      clusters = {
+          "space-x01" = {
+              service_account_whitelist = ["spire:spire-agent"]
+          },
+          "space-x02" = {
+              service_account_whitelist = ["spire:spire-agent"]
+              kube_config_file = "/run/spire/kubeconfigs/space-x02"
+          },
+          "kube-01" = {
+              service_account_whitelist = ["spire:spire-agent"]
+              kube_config_file = "/run/spire/kubeconfigs/kube-01"
+          },
```

Comment:
nit: does this parse with the trailing commas on the JSON? Not sure some parses are strict some are not

Reply:
It worked fine during testing, without remote clusters. We are fine here